core/run.py

```diff
@@ -101,7 +107,7 @@ def run():
     postprocessor = DefaultPostprocessor()
     skill_caller = RestCaller(max_workers=MAX_WORKERS)
     response_selector = ConfidenceResponseSelector()
-    ss_names, ss_urls = zip(*[(annotator['name'], annotator['url']) for annotator in SKILL_SELECTORS])
+    ss_names, ss_urls = zip(*[(annotator['name'], '{}:{}/{}'.format(annotator['host'], annotator['port'], annotator['endpoint'])) for annotator in SKILL_SELECTORS])
```

Comment:
This logic should be implemented for all services, not only annotators (see my suggested code in the previous comment)